111-use-buildah-instead-of-kaniko.md

```diff
@@ -0,0 +1,236 @@
+  annotations:
+    strimzi.io/use-connector-resources: "true"
+spec:
+  version: 4.1.0
+  replicas: 1
+  bootstrapServers: my-cluster-kafka-bootstrap:9093
+  tls:
+    trustedCertificates:
+      - secretName: my-cluster-cluster-ca-cert
+        pattern: "*.crt"
+  config:
+    group.id: connect-cluster
+    offset.storage.topic: connect-cluster-offsets
+    config.storage.topic: connect-cluster-configs
+    status.storage.topic: connect-cluster-status
+    config.storage.replication.factor: -1
+    offset.storage.replication.factor: -1
+    status.storage.replication.factor: -1
+  build:
+    output:
+      type: docker
+      image: my-internal-registry:5000/repository/image:latest
+      additionalBuildOptions:
+        - "--tls-verify=false"
+      additionalPushOptions:
+        - "--quiet"
+    plugins:
+      - name: kafka-connect-file
+        artifacts:
+          - type: maven
+            group: org.apache.kafka
+            artifact: connect-file
+            version: 4.1.0
+```
+
+### Running Buildah in restricted environment
+
+To successfully run Buildah in container on Kubernetes, the Pod needs to have some capabilities.
+The capabilities are `cap_kill`, `cap_setgid`, and `cap_setuid`, which are used in the `build` phase.
+In case that user will use restricted Pod Security profile, Buildah will not work - as these capabilities are dropped in this profile.
+That is similar to Kaniko - which is not able to run in this restricted profile as well.
+Additionally, that is a reason why we will not implement Buildah for OpenShift - and it's described as [one of the rejected alternatives](#only-one-implementation-of-connect-build---buildah-on-both-kubernetes-and-openshift).
+
+## Affected/not affected projects
+
+The one and only affected project is `strimzi-kafka-operator` repository, especially following classes:
+- `KafkaConnectBuild`
+- `ConnectBuildOperator`
+- `KafkaConnectAssemblyOperator`
+
+These classes contain the logic of the Connect Build feature, thus they are affected the most.
+We will need also changes to `DockerOutput` model and to the `KafkaConnect` CRD in order to support two new fields for specifying the additional Buildah options.
+
+## Compatibility
+
+Even though it's change in terms of the tool that builds the images, it will be gated behind the feature gate, so users can adapt to this change through few releases.
+Until the feature gate is GA, we will keep the Kaniko implementation in place, ensuring the backwards compatability.
+The `.spec.build.additionalKanikoOptions` field will be deprecated from the start, however it will be usable until the Buildah feature gate is moved to Beta phase or removed when moving to the `v1` API.
+After that, the `additionalKaninkoOptions` field will be completely ignored and later (with v1 API) it will be removed.
+But we will give users enough time to change between `additionalKanikoOptions` and `additionalOptions` before we reach the Beta phase, so it shouldn't be a problem.
```

Comment:
I guess it has to be replaced by `additionalBuildOptions` and `additionalPushOptions`?